dbms/src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -585,6 +613,23 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                 ErrorCodes::BAD_ARGUMENTS);
     }
 
+    /// MultiversionMergeTree must have version column in primary key. Add it to primary key implicitly otherwise.
+    if (merging_params.mode == MergeTreeData::MergingParams::Multiversion)
+    {
+        auto ast_primary_expr_list = typeid_cast<ASTExpressionList *>(primary_expr_list.get());
+        bool has_version_column_in_pk = false;
+        for (const auto & expr : ast_primary_expr_list->children)
```

Comment:
Use `{ }` for nested `for`, `if` and `if`

Reply:
Does it work if the version if a function of some column?

Reply:
Good point!
Version as last setting in VersionedCollapsingMergeTree(...., version) is an ASTIdentifier, like it was in CollapsingMergeTree. If we have function(version) in primary key, it won't work. But, it doesn't seem to be a problem because we can add version anyway as last field of primary key.